src/middlewares/error.middleware.js

```diff
@@ -0,0 +1,14 @@
+import { ReasonPhrases, StatusCodes } from 'http-status-codes';
+import { ApiError } from '../exceptions/api.error.js';
+
+export const errorMiddleware = (error, req, res, next) => {
+  if (error instanceof ApiError) {
+    const { status, message, errors } = error;
+
+    return res.status(status).send({ message, errors });
+  }
+
+  return res
+    .status(StatusCodes.INTERNAL_SERVER_ERROR)
+    .send(error.message || ReasonPhrases.INTERNAL_SERVER_ERROR);
```

Comment:
The error handling for non-ApiError instances should ensure that `error.message` is a string before sending it in the response. Consider using a default message like `ReasonPhrases.INTERNAL_SERVER_ERROR` if `error.message` is not a string.